### PR TITLE
API and Schema Versioining

### PR DESCRIPTION
**Pull request by govindkrjoshi**

### CODE REVIEW

File: doc/arch/003-api-versioning.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Status
+
+Proposed
+
+## Context
+
+As the application evolves over time, we will
+- add more apis
+- change the behavior exiting apis
```
Comment (pratul) [inline doc/arch/003-api-versioning.md]:
```suggestion
- change the behaviour of existing apis
```

File: doc/arch/003-api-versioning.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- deprecate unused apis
+- add feilds to requests and response schemas
```
Comment (pratul) [inline doc/arch/003-api-versioning.md]:
```suggestion
- add fields to requests and response schemas
```

File: doc/arch/003-api-versioning.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+The api version is derived from the schema version, and is simply the minor version of the schema. 
+Adding a new api is backwards compatible and don't require us to increment the api verion. Any breaking change to the api requires us to publish a new api version. It should be noted that all other endpoints, except for the new breaking changes, should be compatible with the previous version of the API. 
+
+We also maintain the code in the latest api, and update older versions to maintain backwards compatibility.
+## Consequences
+
+- Any change to an existing api, should be forethought.
+- Collorary: Adding new apis is preferrred over updating existing ones.
+- We need to maintain controllers and views for every api version. This can cause code duplication across versions.
+- We need to meaure the number of clients using each version of the api so we can deprecate older apis.
```
Comment (pratul) [inline doc/arch/003-api-versioning.md]:
```suggestion
- We need to measure the number of clients using each version of the api so we can deprecate older apis.
```

**Comment by govindkrjoshi** [comment]
Closing this PR in favor of https://github.com/simpledotorg/simple-server/pull/163 and https://github.com/simpledotorg/simple-server/pull/164

**Pull request closed without merging**